### PR TITLE
Rahul Desai [Node Express Lab]

### PR DESCRIPTION
**Pull request by rd825**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,5 +1,83 @@
...
+                res.status(200).json(post);
+            } 
+            res.status(404).json({message: "The post with the specified ID does not exist."});
+        })
+        .catch(err => {
+            res.status(500).json({error: "The post information could not be retrieved."})
+        })
+})
+
+server.post('/api/posts/', async (req, res) => {
```
Comment (Ta1grr) [inline index.js]: If you're doing one, might as well convert all of them to async/await

File: index.js [hunk 2]
```diff
@@ -1,5 +1,83 @@
...
+    if (req.body.title !== undefined && req.body.contents !== undefined) {
+        try {
+            const postId = await db.insert(req.body);
+            db.findById(postId.id).then(post => res.status(201).json(post)).catch(err => res.status(404).json({message: "The post with the specified ID does not exist."}));
```
Comment (Ta1grr) [inline index.js]: You don't really need to post by Id, it just want you to return an id when you post successfully.

File: index.js [hunk 3]
```diff
@@ -1,5 +1,83 @@
...
+        }
+        catch (err) {
```
Comment (Ta1grr) [inline index.js]: Here's another way to do it.
```
server.post('/api/posts/', (req, res) => {
    const { title, contents } = req.body;
    if (!title || !contents ) {
        res.status(400).json(`{ errorMessage: "Please provide title and contents for the post." }`)
    };

    db.insert({ title, contents }).then(posts => {
        posts.push({ title, contents });
        res.status(201).json(`{Message: "Success"}`);
    })
    .catch(err => res.send(err));  //Could add this instead: { error: "There was an error while saving the post to the database" }
})
```

File: index.js [hunk 4]
```diff
@@ -1,5 +1,83 @@
...
+    }
+})
+
+server.delete('/api/posts/:id', (req,res) => {
+    db.remove(req.params.id).then(count => {
+        if (count) {
+            res.status(200).json(count);
+        }
+        else {
+            res.status(404).json({message: "The post with the specified ID does not exist."});
```
Comment (Ta1grr) [inline index.js]: This could be in your `.catch` instead.

File: index.js [hunk 5]
```diff
@@ -1,5 +1,83 @@
...
+})
+
+server.put('/api/posts/:id', (req,res) => {
+    if (req.body.title !== undefined && req.body.contents !== undefined) {
+        db.update(req.params.id, req.body).then(count => {
+                if (count) {
+                    db.findById(req.params.id).then(post => res.status(200).json(post)).catch(err => res.status(404).json({message: "The post with the specified ID does not exist."}));
+                }
+                else {
+                    res.status(404).json({message: "The post with the specified ID does not exist."});
```
Comment (Ta1grr) [inline index.js]: This should be in your `.catch` instead, no need for an `else` statement

File: index.js (hunk 2 above)
Comment (rd825) [reply index.js]: So the instructions were conflicting. In one place it said just the ID and then lower down in the Endpoint Specs section, it said 
> return the newly created post.

File: index.js (hunk 1 above)
Comment (rd825) [reply index.js]: Think I might stop doing async / await actually. Don't get it as well as the way we've been doing it.

File: index.js (hunk 3 above)
Comment (rd825) [reply index.js]: So throughout this, I was going off the Endpoint Specs section of the instruction that wanted to account for tons of error cases. 

Though I do find your version here a lot more succinct and elegant.

**Pull request closed without merging**